### PR TITLE
Cleanup error handlers

### PR DESCRIPTION
**Pull request by nnethercote**
Mostly by making function naming more consistent. More to do after this, but this is enough for one PR.

r? compiler-errors

### CODE REVIEW

File: src/tools/rust-analyzer/crates/hir-ty/src/layout.rs [hunk 1]
```diff
@@ -110,7 +110,7 @@ struct LayoutCx<'a> {
 impl<'a> LayoutCalculator for LayoutCx<'a> {
     type TargetDataLayoutRef = &'a TargetDataLayout;
 
-    fn delay_bug(&self, txt: String) {
+    fn delayed_bug(&self, txt: String) {
```
Comment (lnicola) [inline src/tools/rust-analyzer/crates/hir-ty/src/layout.rs:113]: CC @Veykril this will be fun to handle.

Comment (nnethercote) [reply src/tools/rust-analyzer/crates/hir-ty/src/layout.rs:113]: That particular commit isn't that important, so if it's a pain I can just remove it.

Comment (Veykril) [reply src/tools/rust-analyzer/crates/hir-ty/src/layout.rs:113]: Isn't that only a problem when we do a sync while the changes haven't been pushed to crates io yet? fwiw we can change the autopublish time to be right before your usual sync times if you want

Comment (lnicola) [reply src/tools/rust-analyzer/crates/hir-ty/src/layout.rs:113]: It is, but it's easy to patch when syncing back, then undo that when we upgrade to the published version.

**Comment by nnethercote** [comment]
I added a few more commits. I'll stop now, I promise :)

File: compiler/rustc_errors/src/lib.rs [hunk 2]
```diff
@@ -996,14 +996,14 @@ impl Handler {
         self.inner.borrow_mut().span_bug(span, msg)
     }
 
-    /// For documentation on this, see `Session::delay_span_bug`.
+    /// For documentation on this, see `Session::span_delayed_bug`.
```
Comment (compiler-errors) [inline compiler/rustc_errors/src/lib.rs:998]: Can you amend this commit with a note that this used to be called `delay_span_bug`? People will probably not like that this is renamed 😓

**Review by compiler-errors — APPROVED** [review]

File: compiler/rustc_errors/src/lib.rs (hunk 2 above)
Comment (nnethercote) [reply compiler/rustc_errors/src/lib.rs:998]: :+1:

**Pull request merged**